hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AbstractFSWAL.java

```diff
@@ -142,7 +142,7 @@
     "hbase.regionserver.wal.slowsync.roll.interval.ms";
   protected static final int DEFAULT_SLOW_SYNC_ROLL_INTERVAL_MS = 60 * 1000; // in ms, 1 minute
 
-  protected static final String WAL_SYNC_TIMEOUT_MS = "hbase.regionserver.wal.sync.timeout";
+  public static final String WAL_SYNC_TIMEOUT_MS = "hbase.regionserver.wal.sync.timeout";
```

Comment:
This is a missing part in our design, usually, if here we get a timeout exception, the only correct way is to abort the region server, as the design of WAL sync, is to succeed or die, there is no 'failure'. It is usually not a big deal is because we set a very large default value here, 5 minutes, usually the WAL system will abort the region server if it can not finish the sync within 5 minutes...

So I think we should throw a special IOException to upper layer, if we get this exception, we abort the region server.

Reply:
@Apache9 ,ok, thank you very much for explanation, I would try to fix the code following your point.

Reply:
I think this should be another issue.

Reply:
@Apache9, I have read the code and have the problem:

Reply:
@Apache9 , I have read the code and have a problem

Reply:
@Apache9 , after I read the code about WAL , I  have a question:
For `AsyncFSWAL`, basically  `WAL.sync` could not throw any exception except that `TimeoutIOException`,
but for `FSHLog`, `WAL.sync` could throw the exception thrown by `ProtobufLogWriter.append` and 
`ProtobufLogWriter.sync ` ,  it just requests the WAL rolling and does not abort the RegionServer, so for  `AsyncFSWAL`,we could abort the RegionServer, but for `FSHLog`, it is not suitable to abort the RegionServer when `WAL.sync` throws the exception other than `TimeoutIOException` , we still need a  way to avoid the situation described by this issue.

Reply:
This is also a problem for the FSHLog implementation. Basically, if the write to HDFS fails, we do not know whether the data has been persistent or not. The implementation for AsyncFSWAL, is to open a new writer and try to write the WAL entries again, and then adding logic in WAL split and replay to deal with duplicate entries. So for FSHLog, if it is not easy to add the same logic with AsyncFSWAL, the correct way is to abort the region server to let the failover logic to detect whether the WAL entries have been persistent or not.

Reply:
@Apache9,  ok,   I think we could open two new jiras , one is for AsyncFSWAL, and the other is to implement the logic of HLog as same as AsyncFSWAL , which is a little more complicated.

Reply:
@Apache9 , I have opened HBASE-27230 and HBASE-27231 for these two problems